sdk/python/sawtooth_sdk/client/encoding.py

```diff
@@ -0,0 +1,302 @@
+                       outputs=None,
+                       payload_encoding=None):
+        """Convenience method which creates a single Transaction from a payload
+        and supplied or default TransactionHeadewr values, before wrapping it
+        in a TransactionList and serializing it. Accepts identical parameters
+        to TransactionEncoder.create().
+
+        Args:
+            payload (bytes): If no payload_encoder is set on the
+                TransactionEncoder, the payload must be byte-encoded
+            batcher_pubkey (string or bytes, optional): Batcher public key
+            dependencies (list of str, optional): Transaction ids that must be
+                committed before this Transaction
+            family_name (str, optional): Name of Transaction Family
+            family_version (str, optional): Version of Transaction Family
+            inputs (list of str, optional): State addresses that Transaction
+                created by this encoder will read from
+            nonce (str, optional): Random string to ensure uniqueness
+            outputs (list of str, optional): State addresses that Transaction
+                created by this encoder will write to
+            payload_encoding (str, optional): The byte-encoding the Transaction
+                Processor should expect
+
+        Raises:
+            TypeError: Raised if a user-supplied required header was not set
+                in either the TransactionEncoder, or during this method call.
+                Required user-supplied headers include:
+                    * family_name
+                    * family_version
+                    * payload_encoding
+                    * inputs
+                    * outputs
+
+        Returns:
+            bytes: a serialized TransactionList
+        """
+        txn = self.create(payload,
+                          batcher_pubkey=batcher_pubkey,
+                          dependencies=dependencies,
+                          family_name=family_name,
+                          family_version=family_version,
+                          inputs=inputs,
+                          nonce=nonce,
+                          outputs=outputs,
+                          payload_encoding=payload_encoding)
+        return self.encode(txn)
+
+
+class BatchEncoder(object):
+    """Stores a private key which can be used to create and sign Batches of
+    Transactions.
+
+    Args:
+        private_key (str or bytes): The private key to sign Batches with
+    """
+    def __init__(self, private_key):
+        self._private_key = private_key
+        self._public_key = signing.get_public_key(private_key)
+
+    def create(self, transactions):
+        """Creates and signs a new Batch message with one or more Transactions.
+        Transactions can be input as Transaction instances or as a serialized
+        TransactionList.
+
+        Args:
+            transactions (bytes or Transaction or list of Transaction):
```

Comment:
The stuff in parentheses precisely outline the expected type(s) of the arguments. In this case a `list` of `Transaction` instances. You do not pluralize the class name.